firestore/src/app/homepage/homepage.component.spec.ts

```diff
@@ -96,7 +106,33 @@ describe('HomepageComponent', () => {
     expect(component).toBeTruthy();
   });
 
-  it('should get resturants when service called', () => {
-    expect(component.restaurants).toBeDefined();
-  })
+  /** 
+   * Tests to see that the #empty-restaurants-container div has not
+   * been rendered. This div is rendered only when no restaurants have been 
+   * retrieved from Firestore (or, in this case, the MockHomepageFirestore 
+   * element).
+   * */
+  it('should call service and get restuarants on init', () => {
+    const emptyRestaurantsDiv = fixture.debugElement.query(By.css("#empty-restaurants-container"));
+
+    expect(emptyRestaurantsDiv).toBeNull();
+  });
+
+  it('should change filters when dialog changes', waitForAsync(() => {
+    let mockDialogRef = jasmine.createSpyObj('MatDialogRef', ['close', 'afterClosed']);
+    mockDialogRef.afterClosed.and.returnValue(of({
+      ...DEFAULT_SORT_DATA, price: 2,
+    }));
+    spyOn(component.dialog, 'open').and.returnValue(mockDialogRef);
+
+    component.openFilterDialog();
+    mockDialogRef.close();
+
+    expect(component.sortingData).toEqual({
```

Comment:
Can we assert the values in the UI directly? that way we're capturing the rendering portion also not just the component state. Ideally sortingData should be private data in the component; sometimes the underscore prefix is used to mark a private variable, but this is discouraged: go/tsjs-style#identifiers

Reply:
Gotcha; Added another test that makes assertions on UI elements!